compiler/rustc_hir_typeck/messages.ftl

```diff
@@ -100,7 +100,14 @@ hir_typeck_lossy_provenance_ptr2int =
 hir_typeck_missing_parentheses_in_range = can't call method `{$method_name}` on type `{$ty_str}`
 
 hir_typeck_never_type_fallback_flowing_into_unsafe =
-    never type fallback affects this call to an `unsafe` function
+    never type fallback affects this {$reason ->
+    [call] call to an `unsafe` function
+    [union_field] union access
+    [deref] raw pointer dereference
+    [path] `unsafe` function
+    [method] call to an `unsafe` method
```

Comment:
Split this into 5 slugs and do:

```
#[derive(Diagnostic)]
enum NeverTypeFallbackFlowingIntoUnsafe {
  #[diag(hir_typeck_never_type_fallback_flowing_into_unsafe_call)]  
  Call,
  ...
}
```

And then

```
hir_typeck_never_type_fallback_flowing_into_unsafe_call = never type fallback affects this call to an `unsafe` function
```

Reply:
I've kept the reason enum and _then_ convert to the diagnostic, cause I find using `errors::*` until the literal "emit" somewhat weird. But LMK if I should just return the diagnostic from the "find unsafe infer vars" function.